### PR TITLE
Alternative-Quest scanning support

### PR DESCRIPTION
**Pull request by 123FLO321**
## Description
Add support for AR Quest Scanning

## How Has This Been Tested?
Totally untested 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: Sources/RealDeviceMap/Modell/Pokestop.swift [hunk 1]
```diff
@@ -57,7 +63,28 @@ class Pokestop: JSONConvertibleObject, WebHookEvent, Hashable {
                 "updated": questTimestamp!,
                 "pokestop_name": name ?? "Unknown",
                 "ar_scan_eligible": arScanEligible ?? 0,
-                "pokestop_url": url ?? ""
+                "pokestop_url": url ?? "",
+                "is_ar": false
```
Comment (Petap0w) [inline Sources/RealDeviceMap/Modell/Pokestop.swift:67]: Maybe as part of the global "renaming of a few things" to match the idea that we want to display set A or set B of quests (and not only AR ones of set B), this addition to the webhook sent should be more like `"with_ar": true` here (to reflect this quest is available when holding an AR)?

**Comment by TurtIeSocks** [comment]
@123FLO321 just to clarify and make sure I'm reading this correctly, the alternative quest columns store the quests that are scanned when the device is not holding an AR quest?

**Comment by TurtIeSocks** [comment]
Well if anyone is testing this, ReactMap now supports it, though it's untested since I'm not running this PR:
https://github.com/WatWowMap/ReactMap/pull/187

**Comment by Fabio1988** [comment]
> @123FLO321 ~just to clarify and make sure I'm reading this correctly, the alternative quest columns store the quests that are scanned when the device is not holding an AR quest?~ After doing some more reading, it seems the opposite is true?

I think your first strikethrough sentence is the right way.

actually RDM is saving quests with holding ar quest in inventory.
when RDM will introduce the alternative quest it will save quests which were received NOT holding AR-quest.

File: resources/webroot/static/data/de.json [hunk 2]
```diff
@@ -3917,6 +3917,7 @@
         "filter_pokestop_lure_4": "Magnet-Lockmodul",
         "filter_pokestop_lure_5": "Regen-Lockmodul",
         "filter_pokestop_ar_only": "Nur AR geeignete Pokestops",
+        "filter_quest_show_ar": "AR Quests Anzeigen",
```
Comment (Fabio1988) [inline resources/webroot/static/data/de.json:3920]: Maybe change it to "Alternative Quests anzeigen" ?

File: resources/webroot/static/data/en.json [hunk 3]
```diff
@@ -3924,6 +3924,7 @@
         "filter_pokestop_lure_4": "Magnetic Lure",
         "filter_pokestop_lure_5": "Rainy Lure",
         "filter_pokestop_ar_only": "Only AR Eligible Pokestops",
+        "filter_quest_show_ar": "Show AR Quests",
```
Comment (Fabio1988) [inline resources/webroot/static/data/en.json:3927]: also here, change it to "Show Alternative Quests"

**Comment by Fabio1988** [comment]
We should be careful with naming.... It's confusing to name it "Ar-quest-condition"

File: resources/webroot/index.js.mustache [hunk 4]
```diff
@@ -2501,6 +2524,7 @@ function getPokestopPopupContent (pokestop) {
       conditionsString += ')'
     }
 
+    content += '<b>AR-Quest in Inventory:</b><br>'
```
Comment (Fabio1988) [inline resources/webroot/index.js.mustache:2527]: Maybe make it translatable? :)

**Comment by Fabio1988** [comment]
on editing any instance the field "quest mode" is showing up on inittial load, should only be shown if it's of type "quest"

**Comment by Fabio1988** [comment]
if you already run that PR please execute the following SQL-query to prevent migration errors

```
ALTER TABLE pokestop
    DROP INDEX `ix_alternative_quest_alternative_quest_pokemon_id`,
    DROP INDEX `ix_alternative_quest_reward_type`,
    DROP INDEX `ix_alternative_quest_item_id`,
    DROP COLUMN `alternative_quest_pokemon_id`,
    DROP COLUMN `alternative_quest_reward_type`,
    DROP COLUMN `alternative_quest_item_id`,
    DROP COLUMN `alternative_quest_type`,
    DROP COLUMN `alternative_quest_timestamp`,
    DROP COLUMN `alternative_quest_target`,
    DROP COLUMN `alternative_quest_conditions`,
    DROP COLUMN `alternative_quest_template`;
ALTER TABLE pokestop
    DROP COLUMN `alternative_quest_rewards`;

```

it is now rebased on development.. and please change the DB_VERSION in table `metadata` to `69`, because this PR contains migration 70.

**Pull request merged**